### PR TITLE
New updates with Webpack 4 & Add mini-css-extract-plugin test

### PR DESCRIPTION
**Pull request by weilinzung**
Big updates:

- UPDATE to WEBPACK 4
- ADD Mini-CSS-Extract-plugin test 
- FIX all compatibility issues

#89

### CODE REVIEW

File: test/unit/transformers/moduleNotFound.spec.js [hunk 1]
```diff
@@ -34,16 +33,4 @@ it('Sets the appropiate type', () => {
 it('Ignores other errors', () => {
   const error = { name: 'OtherError' };
   expect(moduleNotFound(error)).toEqual(error);
-});
-
-it('Sets the correct message with a RequireContextDependency', () => {
```
Comment (christophehurpeau) [inline test/unit/transformers/moduleNotFound.spec.js]: why did you remove this test ?

File: src/transformers/babelSyntax.js [hunk 2]
```diff
@@ -11,15 +11,19 @@ function cleanStackTrace(message) {
...
+    .replace(/^Module build failed.*:\s/, 'Syntax Error: ')
+    // remove mini-css-extract-plugin loader tracing errors
+    .replace(/^Syntax Error: ModuleBuildError:.*:\s/, '')
+    // remove babel extra wording
+    .replace('SyntaxError: ', '');
 }
 
 function isBabelSyntaxError(e) {
-  return e.name === 'ModuleBuildError' &&
+  return e.name === 'ModuleBuildError' || e.name === 'ModuleBuildError' &&
```
Comment (christophehurpeau) [inline src/transformers/babelSyntax.js]: are you testing the same string ?

File: package.json [hunk 3]
```diff
@@ -33,22 +33,28 @@
...
+    "expect": "^24.8.0",
     "jest": "^18.1.0",
     "memory-fs": "^0.4.1",
-    "webpack": "^2.2.1"
+    "mini-css-extract-plugin": "^0.6.0",
+    "node-sass": "^4.12.0",
+    "sass": "^1.20.1",
+    "sass-loader": "^7.1.0",
+    "style-loader": "^0.23.1",
+    "webpack": "^4.31.0"
```
Comment (christophehurpeau) [inline package.json]: we could also change the peerdependency and make a breaking change

File: test/integration.spec.js [hunk 4]
```diff
@@ -110,6 +114,15 @@ it('integration : babel syntax error', async() => {
   ]);
 });
 
+it('integration : mini CSS extract plugin babel error', async() => {
+
+  const logs = await executeAndGetLogs('./fixtures/mini-css-extract-babel-syntax/webpack.config');
+  const clean_logs = logs.toString().replace(/\"/g, ""); //<- double quotes issue with slash
```
[the remark this answers is not in the document]
Comment (weilinzung) [reply test/integration.spec.js]: the logs that I put it in always has the extra `\` before `"`, you may want to try it without the replace and you will know what I meant.

File: src/transformers/babelSyntax.js (hunk 2 above)
Comment (weilinzung) [reply src/transformers/babelSyntax.js]: `e.name === 'ModuleBuildError' && e.message.indexOf('SyntaxError') >= 0;` this one only works both exist wich casue on SCSS errors can't catch it.

So I am adding if no `e.message` which still can catch `'ModuleBuildError'`.

File: package.json (hunk 3 above)
Comment (weilinzung) [reply package.json]: I am only doing for the new feature. Once I update all other old packages would have too many errors, I don't want to touch them now.

File: test/fixtures/success/webpack.config.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 const FriendlyErrorsWebpackPlugin = require('../../../index');
 
 module.exports = {
+	mode: 'development',
```
[the remark this answers is not in the document]
Comment (weilinzung) [reply test/fixtures/success/webpack.config.js]: It is two tabs for me.

[the remark this answers is not in the document]
Comment (weilinzung) [reply test/fixtures/success/webpack.config.js]: fixed!

**Comment by christophehurpeau** [comment]
Thank you @

**Comment by weilinzung** [comment]
@christophehurpeau when would be released?

**Comment by christophehurpeau** [comment]
@weilinzung sorry for the delay, released as 2.0.0-beta.1 :)
